wallets/provider-xdefi/src/index.ts

```diff
@@ -77,7 +76,11 @@ export const subscribe: Subscribe = ({
      */
     updateChainId(chainId);
     connect(network);
-  });
+  };
+  const eth = chooseInstance(instance, meta, Networks.ETHEREUM);
+  eth?.on('chainChanged', handleChainChanged);
+
+  return () => eth?.off('chainChanged', handleChainChanged);
```

Comment:
```suggestion
  return () => { eth?.off('chainChanged', handleChainChanged); }
```

I don't know what `off` returns but for making sure it won't break anything in future, let's stick to the `void` which is defined as return type for cleanup.

Reply:
fixed.